File: Tarea 05/CalcularIMC.m
#Definición de las categorías
bajoPeso = "Bajo peso";
pesoNormal = "Peso normal";
sobrePeso = "Sobrepeso";

#Bucle principal
while true
    disp("------------------------------");
    fprintf("\n");

    #Mostrar opciones
    disp("Opciones");
    disp("1. Calcular IMC y mostrar resultados");
    disp("2. Leer información del archivo");
    disp("3. Borrar información del archivo");
    disp("4. Salir del programa");

    #Leer opción del usuario
    opcion = input("Ingrese la opción deseada: ");

    #Validar la opción
    if not (1<= opcion <=4)
        disp("Opción no válida. Intente de nuevo.");
        continue;
    end

    #Ejecutar la acción según la opción
    switch opcion
        case 1
            #Calcular IMC y mostrar resultados
            nombre = input("Ingrese su nombre: ", "s");
            peso = input("Ingrese su peso en kilogramos: ");
            altura = input("Ingrese su altura en metros: ");

            imc = peso / (altura^2);

            #Clasificar el IMC
            if imc < 18.5
                categoria = bajoPeso;
            elseif imc < 25
                categoria = pesoNormal;
            else
                categoria = sobrePeso;
            end

            #Mostrar resultados
            fprintf("\nNombre: %s\n", nombre);
            fprintf("IMC: %.2f\n", imc);
            fprintf("Categoría: %s\n", categoria);

            #Guardar en archivo
            archivo = fopen("imc.txt", "a");
            fprintf(archivo, "Nombre: %s\nIMC: %.2f\nCategoría: %s\n\n", nombre, imc, categoria);
            fclose(archivo);

        case 2
            #Leer información del archivo
            if exist("imc.txt", "file")
                disp("Contenido de imc.txt:");
                tipo = fopen("imc.txt", "r");
                while ~feof(tipo)
                    linea = fgetl(tipo);
                    disp(linea);
                end
                fclose(tipo);
            else
                disp("No existe el archivo imc.txt.");
            end

        case 3
            #Borrar información del archivo
            if exist("imc.txt", "file")
                delete("imc.txt");
                disp("Archivo imc.txt eliminado.");
            else
                disp("No existe el archivo imc.txt.");
            end

        case 4
            #Salir del programa
            disp("¡Gracias por usar el programa!");
            break;
    end
end

File: Tarea 05/CalcularIMC.py
import psycopg2

# Conexión a PostgreSQL
conn = psycopg2.connect(
    dbname="postgres",
    user="postgres",
    password="koala",
    host="localhost",
    port="5432"
)
cursor = conn.cursor()

# Definición de categorías
bajo_peso = "Bajo peso"
peso_normal = "Peso normal"
sobre_peso = "Sobrepeso"

def guardar_en_archivo(nombre, imc, categoria):
    with open("imc.txt", "a") as f:
        f.write(f"Nombre: {nombre}\nIMC: {imc:.2f}\nCategoría: {categoria}\n\n")

def leer_archivo():
    try:
        with open("imc.txt", "r") as f:
            print("Contenido de imc.txt:")
            for linea in f:
                print(linea.strip())
    except FileNotFoundError:
        print("No existe el archivo imc.txt.")

def borrar_archivo():
    import os
    if os.path.exists("imc.txt"):
        os.remove("imc.txt")
        print("Archivo imc.txt eliminado.")
    else:
        print("No existe el archivo imc.txt.")

def borrar_base_datos():
    cursor.execute("DELETE FROM imc")
    conn.commit()
    print("Registros de la base de datos eliminados.")

while True:
    print("------------------------------\n")
    print("Opciones")
    print("1. Calcular IMC y mostrar resultados")
    print("2. Leer información del archivo")
    print("3. Borrar información del archivo y base de datos")
    print("4. Salir del programa")

    opcion_str = input("Ingrese la opción deseada: ")
    if not opcion_str.isdigit() or int(opcion_str) not in range(1,5):
        print("Opción no válida. Intente de nuevo.")
        continue

    opcion = int(opcion_str)

    if opcion == 1:
        nombre = input("Ingrese su nombre: ")
        peso = float(input("Ingrese su peso en kilogramos: "))
        altura = float(input("Ingrese su altura en metros: "))
        imc = peso / (altura ** 2)

        if imc < 18.5:
            categoria = bajo_peso
        elif imc < 25:
            categoria = peso_normal
        else:
            categoria = sobre_peso

        print(f"\nNombre: {nombre}")
        print(f"IMC: {imc:.2f}")
        print(f"Categoría: {categoria}")

        guardar_en_archivo(nombre, imc, categoria)

        cursor.execute(
            "INSERT INTO imc (nombre, imc, categoria) VALUES (%s, %s, %s)",
            (nombre, imc, categoria)
        )
        conn.commit()

    elif opcion == 2:
        leer_archivo()

    elif opcion == 3:
        borrar_archivo()
        borrar_base_datos()

    elif opcion == 4:
        print("¡Gracias por usar el programa!")
        break

cursor.close()
conn.close()
